symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/BdkExponentialFunction.java

```diff
@@ -0,0 +1,28 @@
+package com.symphony.bdk.core.util;
+
+import com.symphony.bdk.core.config.model.BdkRetryConfig;
+import io.github.resilience4j.core.IntervalFunction;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BdkExponentialFunction {
+
+    public static IntervalFunction ofExponentialBackoff(BdkRetryConfig retryConfig) {
```

Comment:
it seems resilience4j is exposing a function already: https://resilience4j.readme.io/docs/retry#use-a-custom-intervalfunction
do we do something different here?

looks like the getMaxIntervalMillis is what we do differently

Reply:
but in a sense since we have maxAttempts(bdkRetryConfig.getMaxAttempts())
don't we already limit the max interval duration?

Reply:
I think the `maxIntervalMillis` is needed to limit the interval in some case that user want to have many many attempts but not exploding the interval by the multiplier.